### PR TITLE
Bugfix/accuracy tests

### PR DESCRIPTION
**Pull request by leoTiez**

### CODE REVIEW

File: RCNN_tf.py [hunk 1]
```diff
@@ -23,10 +22,11 @@
 PRECISION_TF = tf.float32
 PRECISION_NP = np.float32
 PADDING_LIST = ['SAME', 'VALID']
+initializer = tf.contrib.layers.variance_scaling_initializer()
```
Comment (bt-s) [inline RCNN_tf.py]: Good. Maybe we could capitalize it macro style, but this is also fine!

File: RCNN_tf.py [hunk 2]
```diff
@@ -104,97 +81,80 @@ def rcl(input_data, num_input_chans, num_filter, filter_shape, num_of_data,
...
-                depth_radius=num_norm_feat_maps,
-                bias=1,
-                alpha=alpha,
-                beta=beta,
-                name=name + '_lrn'
-            )
+        return x, recurrent_states, output
 
-        return output
+    # TODO I am not sure why but this learns quicker. Investigate?
```
Comment (bt-s) [inline RCNN_tf.py:133]: It probably has to do with the order in which Tensorflow calls the functions in the while loop.
We could investigate it, but if you have seen that it is faster I think we could just leave it as it is.

File: RCNN_tf.py [hunk 3]
```diff
@@ -104,97 +81,80 @@ def rcl(input_data, num_input_chans, num_filter, filter_shape, num_of_data,
         padding='SAME'
     )
 
-    if "CPU" in processing_unit.upper():
```
Comment (bt-s) [inline RCNN_tf.py:107]: Nice catch that we do not have to specify anymore whether we are using a GPU or a CPU!

File: RCNN_tf.py [hunk 4]
```diff
@@ -203,13 +163,12 @@ def convolutional_layer(input_data, num_input_chans, num_filter, filter_shape,
         output       (tf.Tensor): output tensor of the conv layer
     """
     conv_filter_shape = [filter_shape[0], filter_shape[1], num_input_chans,
-            num_filter]
+                         num_filter]
     strides = [1, stride[0], stride[1], 1]
 
     initializer = tf.contrib.layers.variance_scaling_initializer()
```
Comment (bt-s) [inline RCNN_tf.py]: Do we need to declare the initializer again over here? Can we not just declare it globally?

File: load_data.py [hunk 5]
```diff
@@ -65,17 +79,22 @@ def data_loader(dataset, file_name, dtype):
         with open(str(DATA_DIR + '/cifar-10-batches-py/' + file_name), 'rb') as fo:
             data_dict = cPickle.load(fo)
 
-        data = preprocess_cifar_imgs_(np.asarray(data_dict['data'], dtype=dtype))
+        data = preprocess_cifar_imgs_(np.asarray(data_dict['data'], dtype=dtype),
+                                      dtype=dtype,
```
Comment (bt-s) [inline load_data.py:83]: Is this ```dtype=dtype``` necessary?

File: load_data.py [hunk 6]
```diff
@@ -65,17 +79,22 @@ def data_loader(dataset, file_name, dtype):
...
         labels = to_categorical(np.asarray(data_dict['labels'], dtype=dtype),
                 num_classes=10)
 
     elif dataset == "cifar100":
         with open(str(DATA_DIR + '/cifar-100-python/' + file_name), 'rb') as fo:
             data_dict = cPickle.load(fo)
 
-        data = preprocess_cifar_imgs_(np.asarray(data_dict['data'], dtype=dtype))
+        data = preprocess_cifar_imgs_(np.asarray(data_dict['data'], dtype=dtype),
+                                      dtype=dtype,
```
Comment (bt-s) [inline load_data.py:94]: What about this Is this ```dtype=dtype```; is that also necessary?

**Review by bt-s — APPROVED** [review]
This looks very solid: I like the implementation!

**Review by adrianc-a — APPROVED** [review]
lgtm

File: RCNN_tf.py (hunk 1 above)
Comment (leoTiez) [reply RCNN_tf.py]: No, that's true. Thanks for pointing it out!

File: RCNN_tf.py (hunk 2 above)
Comment (leoTiez) [reply RCNN_tf.py:133]: Yea think so, too. For the moment we can leave it as it is

File: RCNN_tf.py (hunk 4 above)
Comment (leoTiez) [reply RCNN_tf.py]: Yea, you are right, I totally forgot that line. I will remove that!

File: load_data.py (hunk 5 above)
Comment (leoTiez) [reply load_data.py:83]: Yes, it is necessary for converting to grayscale

File: load_data.py (hunk 6 above)
Comment (leoTiez) [reply load_data.py:94]: Same, necessary for the conversion to grayscale

**Comment by leoTiez** [comment]
solves #13 and #24

**Pull request merged**